examples/mem-alloc-test/src/main.rs

```diff
@@ -0,0 +1,275 @@
+     */
+
+    let avg_thread_latency = tot_time_ns as f64 / count as f64;
+
+    let shared_mutex_clone_2 = Arc::clone(&shared_mutex_clone);
+    calculate_and_print_stat(shared_mutex_clone_2, tid, &memsize, avg_thread_latency);
+
+
+}
+
+fn spawn_threads(thread_count: i32) {
+    let mut handles = vec![];
+    let shared_variable = Arc::new(Mutex::new(Counters{
+            avg_duration_large_thread:0.0,
+            avg_duration_medium_thread:0.0,
+            avg_duration_small_thread:0.0,
+            global_average:0.0}));
+
+    let pair = Arc::new((Mutex::new(false), Condvar::new()));
+    let (mut num_small_threads, mut num_medium_threads, mut num_large_threads) = (0, 0, 0);
+    for i in 0..thread_count {
+        let shared_mutex_clone = Arc::clone(&shared_variable);
+        // Spawn a thread that waits until the condition is met
+        let pair_clone = Arc::clone(&pair);
+        let mut memtype;
+
+        match i % 3 {
+            0 =>{
+                    memtype = MemSize::Large;
+                    num_large_threads += 1;
+                },
+            1 =>{
+                    memtype = MemSize::Medium;
+                    num_medium_threads += 1;
+                },
+            2 =>{
+                    memtype = MemSize::Small;
+                    num_small_threads += 1;
+                },
+            _ => return
+        };
+
+        let handle = thread::spawn(move || {
+            worker_thread(i, shared_mutex_clone, memtype, pair_clone);
+        });
+        handles.push(handle);
+    }
+
+    /* Start all the threads */
+    wakeup_all_child_threads(pair);
+
+    /* Wait for all threads to finish */
+    for handle in handles {
+        handle.join().unwrap();
+    }
+
+    /* Calculate final means */
+    let mut data = shared_variable.lock().unwrap();
+    data.avg_duration_large_thread = data.avg_duration_large_thread / num_large_threads as f64;
+    data.avg_duration_medium_thread = data.avg_duration_medium_thread / num_medium_threads as f64;
+    data.avg_duration_small_thread = data.avg_duration_small_thread / num_small_threads as f64;
+    data.global_average =
+        (data.avg_duration_small_thread + data.avg_duration_medium_thread + data.avg_duration_large_thread) /
+        (num_large_threads + num_medium_threads + num_small_threads ) as f64;
+    println!("{},{},{},{},{}", thread_count, data.avg_duration_small_thread,
+             data.avg_duration_medium_thread,
+             data.avg_duration_large_thread,
+             data.global_average);
+
+
+}
+
+fn get_num_processors() -> usize {
+    num_cpus::get()
+}
+
+/* If there are n processors available, we will record the numbers with,2n threads,
+ * then n threads, then n/2 and so on.
+ */
+fn start_tests()
+{
+    println!("NUM_THREADS,LATENCY_SMALL_THREADS,LATENCY_MEDIUM_THREADS,LATENCY_LARGE_THREADS,GLOBAL_AVERAGE");
+    let mut num_processors = get_num_processors();
+    num_processors = 4;
```

Comment:
`num_processors` is being hard-coded to 4, even though we called `get_num_processors()` on the previous line. Was this intentional?

Reply:
No it isn't. I will remove this.